txpipe/rail_pz.py

```diff
@@ -57,20 +65,66 @@ def run(self):
         cls = Estimator._find_subclass(self.config["class_name"])
         estimator = cls(base_config, run_dict)
 
+        training_data = load_training_data(
+            self.get_input(self.training_tag),
+            "hdf5",
+            "photometry"
+        )
+
         # Run the main training phase
-        estimator.inform()
+        estimator.inform(training_data)
 
         # If there is any kind of testing/validation to be run we could
         # do so here.
 
         # Afterwards, save the model.  This assumes that estimator
         # classes can be pickled, which is true for now but see the
         # issue opened on th RAIL repo
-        with self.open_output("photoz_trained_model", wrapper=True) as output:
+        with self.open_output(self.model_tag, wrapper=True) as output:
             output.write(estimator)
 
 
-class PZRailEstimate(PipelineStage):
+class PZRailTrainLens(PZRailTrainSource):
+    """Train a photo-z model using RAIL.
+
+    The Redshift Assessment Infrastructure Layers (RAIL) library provides a uniform
+    interface to DESC photo-z code.
+
+    TXPipe uses RAIL across several different pipeline stages.
+    This stage, which would normally be run first, uses a training set (e.g. of
+    spectroscopic redshifts) to train and save an Estimator object that a later
+    stage can use to measure redshifts of the survey sample.
+    """
+    name = "PZRailTrainLens"
+    training_tag = "photoz_lens_training"
+    testing_tag = "photoz_lens_testing"
+    model_tag = "photoz_lens_model"
+
+    inputs = [
+        (training_tag, DataFile),
+        (testing_tag, DataFile),
+    ]
+
+    outputs = [(model_tag, PickleFile)]
+
+class PZRailTrainLensFromSource(PipelineStage):
```

Comment:
this feels a little odd, maybe it's the name... like why is it not trainsourcefromlens? also, what if we only run part of 3x2, ie clustering only? (maybe that's a larger question)

Reply:
Yeah, good idea, I've added the reverse stage SourceFromLens